partiql-lang/src/main/kotlin/org/partiql/lang/syntax/impl/PartiQLPigVisitor.kt

```diff
@@ -813,7 +852,8 @@ internal class PartiQLPigVisitor(val customTypes: List<CustomType> = listOf(), p
      *
      */
 
-    override fun visitFromClause(ctx: PartiQLParser.FromClauseContext) = visit(ctx.tableReference(), PartiqlAst.FromSource::class)
+    override fun visitFromClause(ctx: PartiQLParser.FromClauseContext) =
+        visit(ctx.tableReference()) as PartiqlAst.FromSource
```

Comment:
The `PartiQLPigVisitor.visit()` call in the original code uses reflection to perform a cast in the result, compared to the regular cast on the result of `AbstractParseTreeVisitor.visit()` in the replacement code.  Reflection is expensive and better be avoided. 
     
The verbosity of the two approaches is comparable, while the internals of `visit()` do not encapsulate much action.